Dockerfile

```diff
@@ -1,17 +1,34 @@
-FROM --platform=linux/amd64 registry.fedoraproject.org/fedora-minimal:40
+FROM nixos/nix:2.22.0 AS build-stage
+ARG TARGETARCH
 
-RUN microdnf install -y curl findutils gzip iputils less postgresql procps shadow-utils tar time which \
-    && microdnf clean all
+RUN tee -a /etc/nix/nix.conf <<EOF
+experimental-features = nix-command flakes
+filter-syscalls = false
+EOF
+
+WORKDIR /app
+
+COPY flake.nix flake.lock  ./
+COPY development ./development
+
+# build the nix environment once as an early docker layer
+RUN nix develop -c true
+
+# copy the go modules and download em
+COPY go.mod go.sum ./
+RUN nix develop -c go mod download
+
+# copy everything else and build the project
+COPY . ./
+RUN nix develop -c make release/server_linux_$TARGETARCH
+
+FROM buildpack-deps:bullseye AS build-release-stage
```

Comment:
This drops a lot of the debugging tools that have saved our butts in prod before:

```
RUN microdnf install -y curl findutils gzip iputils less postgresql procps shadow-utils tar time which \
    && microdnf clean all
```

Some of these might be built-in `bullseye` but it would suck if they weren't there when we ran into problems.

Reply:
I was thinking that we'd want to use the evil-ssh-pod for that, but I am not super experienced with the DL deployment so if you find them useful in the pod I'm totally fine with that, it just means a bigger container but whatever

Reply:
Updated to leave these in